File: backend/routes/patientRoutes.js
const express = require('express');
const router = express.Router();
const {
  updatePatientProfile,
  getPatientProfile,
  getAllPatients
} = require('../controllers/patientController');
const { protect } = require('../middleware/authMiddleware');

router.route('/profile')
  .get(protect, getPatientProfile)
  .put(protect, updatePatientProfile);

router.get('/all', protect, getAllPatients);

module.exports = router;

File: backend/routes/patient_routes.py
from flask import Blueprint
from controllers.patient_controller import (
    get_patient_profile,
    update_patient_profile,
    get_all_patients
)
from middleware.auth_middleware import protect

patient_bp = Blueprint("patient", __name__, url_prefix="/patient")

# GET & PUT /patient/profile
patient_bp.route("/profile", methods=["GET"])(protect(get_patient_profile))
patient_bp.route("/profile", methods=["PUT"])(protect(update_patient_profile))

# GET /patient/all
patient_bp.route("/all", methods=["GET"])(protect(get_all_patients))
